### PR TITLE
Add API functional tests for ApplyCouponToCart functionality

### PR DESCRIPTION
**Pull request by dmytro-ch**
### Fixed Issues (if relevant)

1. magento/graphql-ce#479: [Test Coverage] 'ApplyCouponToCart' functionality

### Manual testing scenarios (*)

N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/ApplyCouponToCartTest.php [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+    /**
+     * @var QuoteIdToMaskedQuoteIdInterface
+     */
+    private $quoteIdToMaskedId;
+
+    /**
+     * @var CouponResourceInterface
+     */
+    protected $couponResource;
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/ApplyCouponToCartTest.php]: Please, check and remove all of unused dependencies like this

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/ApplyCouponToCartTest.php [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+            ],
+        ];
+    }
+
+    /**
+     * @param string $couponCode
+     * @param string $sku
+     * @throws \Exception
+     */
+    private function excludeProductPerCoupon(string $couponCode, string $sku)
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/ApplyCouponToCartTest.php]: Could we move this `set-up` to fixture file (with rollback)?

Example:
```
/**
     * @magentoApiDataFixture Magento/Customer/_files/customer.php
     * @magentoApiDataFixture Magento/Catalog/_files/product_simple.php
     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/customer/create_empty_cart.php
     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/add_simple_product.php
     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/set_new_shipping_address.php
     */
```

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/ApplyCouponToCartTest.php [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+     */
+    public function testApplyExpiredCoupon()
+    {
+        $couponCode = '2?ds5!2d';
+        $this->quoteResource->load(
+            $this->quote,
+            'test_order_with_simple_product_without_address',
+            'reserved_order_id'
+        );
+        $maskedQuoteId = $this->quoteIdToMaskedId->execute((int)$this->quote->getId());
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/ApplyCouponToCartTest.php]: Looks like we can avoid manipulating with QuoteResource (etc..) if we will reuse current fixtures
```
/**
     * @magentoApiDataFixture Magento/Customer/_files/customer.php
     * @magentoApiDataFixture Magento/Catalog/_files/product_simple.php
     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/customer/create_empty_cart.php
     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/add_simple_product.php
     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/set_new_shipping_address.php
     */
```

**Review by naydav — CHANGES_REQUESTED** [review]

**Comment by naydav** [comment]
@dmytro-ch 
I finished one more related example
https://github.com/magento/graphql-ce/pull/497/files

**Comment by dmytro-ch** [comment]
Hello @naydav,
thank you for reviewing this PR.
I've applied the changes according to the recommendations.

Thank you!

**Pull request merged**